pkg/build/registry/buildlog/rest.go

```diff
@@ -186,6 +186,12 @@ func (r *REST) Get(ctx apirequest.Context, name string, opts runtime.Object) (ru
 		// until they have all run, unless we see one of them fail.
 		// If we aren't following the logs, we will run through all the init containers exactly once.
 		for waitForInitContainers {
+			select {
+			case <-ctx.Done():
+				glog.V(4).Infof("timed out while iterating on build init containers for build pod %s/%s", build.Namespace, buildPodName)
+				return
+			default:
+			}
```

Comment:
we shouldn't log that we're iterating through the containers if we aren't going to iterate through the containers because the context timed out.